src/pc/PlatformDeviceControl.c

```diff
@@ -671,36 +672,38 @@ int tcpipPlatformConnect(const char *devPathRead, const char *devPathWrite, void
 
     struct sockaddr_in serv_addr = { 0 };
 
-    size_t len = strlen(devPathWrite);
-    if (!len)
+    const size_t maxlen = 255;
+    size_t len = strnlen(devPathWrite, maxlen + 1);
+    if (len == 0 || len >= maxlen + 1)
         return X_LINK_PLATFORM_INVALID_PARAMETERS;
-    char *const devPathWriteBuff = (char *)malloc(len + 1);
-    if (!devPathWriteBuff)
+    char *const serv_ip = (char *)malloc(len + 1);
+    if (!serv_ip)
         return X_LINK_PLATFORM_ERROR;
```

Comment:
Could put the array onto stack at maxlen (255/256), but lgtm as well

Reply:
Unfortunately this also doesn't work:
```cpp
    const size_t maxlen = 255;
    char serv_ip[maxlen];
```
https://github.com/luxonis/XLink/actions/runs/2273126779

Could have put it hardcoded (or with a define), but will leave as is...